gidgethub/actions.py

```diff
@@ -43,3 +42,30 @@ def command(cmd: str, data: str = "", **parameters: str) -> None:
     )
     cmd_parts.append(f"::{data}")
     print("".join(cmd_parts))
+
+
+_DELIMITER = "__GIDGETHUB_DELIMITER__"
+
+
+def setenv(name: str, value: str) -> None:
+    """Create or update an environment variable.
+
+    The change applies to this action and future actions running in the job.
+    """
+    # https://github.com/actions/toolkit/blob/af821474235d3c5e1f49cee7c6cf636abb0874c4/packages/core/src/core.ts#L35-L53
+    os.environ[name] = value
+    # https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#multiline-strings
+    write_value = f"{name}<<{_DELIMITER}{os.linesep}{value}{os.linesep}{_DELIMITER}"
+    with open(os.environ["GITHUB_ENV"], "a", encoding="utf-8") as file:
+        file.write(write_value + os.linesep)
+
+
+def addpath(path: Union[str, "os.PathLike[Any]"]) -> None:
```

Comment:
`mypy` complains about `os.PathLike`:
```console
gidgethub/actions.py:63: error: Missing type parameters for generic type "_PathLike"
gidgethub/actions.py:63: note: Subscripting classes that are not generic at runtime may require escaping, see https://mypy.readthedocs.io/en/latest/common_issues.html#not-generic-runtime
```
I did some digging into this with the help of the link provided and found out that `os.PathLike` needs to be subscripted and this will require escaping it during runtime. According to the link provided, there are multiple ways of doing this:

- Using `from __future__ import annotations`
- String literal types: `"os.PathLike[Any]"`
- Using `mypy.TYPE_CHECKING`

I have used string literal type here but if you prefer any other way let me know 😄

Reply:
While I would prefer `from __future__ import annotations`, that would require dropping Python 3.6 support, so the string literal seems fine to me.

Reply:
Although the type variable represents the return type, not the argument type.

```suggestion
def addpath(path: Union[str, "os.PathLike[str]"]) -> None:
```

Reply:
Ah, right. Got it, thanks!